### ISSUE TITLE
Translation Contributions: Help Us Translate DiaryVault into Your Language 

### ISSUE THREAD

**Issue by SankethBK** [SankethBK/diaryvault#212]
### Steps to Add a New Language Translation

1. **Get the Language Code**:

   - Visit [Flutter's GlobalMaterialLocalizations documentation](https://api.flutter.dev/flutter/flutter_localizations/GlobalMaterialLocalizations-class.html) to find the language code for your target language.
   - For example:
     - English: `en`
     - Spanish: `es`
     - French: `fr`

2. **Create a `.arb` File**:

   - Navigate to the `lib/l10n/` directory in the project.
   - Create a new `.arb` file named `intl_<language_code>.arb`.
     - For example, if you are contributing Spanish translations, your filename would be `intl_es.arb`.

3. **Translate Messages**:

   - Use the template file `intl_en.arb` as a reference.
   - Inside this file, you will find key-value pairs where the **keys** should remain the same.
   - Translate the corresponding **values** into your target language in your new `.arb` file.
     - **Example**:
       - In `intl_en.arb`:
         ```json
         {
           "appTitle": "DiaryVault",
           "welcomeMessage": "Welcome to DiaryVault!"
         }
         ```
       - In `intl_es.arb`:
         ```json
         {
           "appTitle": "DiaryVault",
           "welcomeMessage": "¡Bienvenido a DiaryVault!"
         }
         ```

4. **Install Flutter Intl Extension (Optional but Recommended)**:

   - Install the [Flutter Intl](https://marketplace.visualstudio.com/items?itemName=localizely.flutter-intl) extension for Visual Studio Code.
   - This extension helps with generating boilerplate code and keeps localization files in sync.

5. **Regenerate Localization Files**:

   - If you're using the Flutter Intl extension, it should automatically generate the necessary files.
   - If not, run the following command in your terminal to regenerate localization files:
     ```bash
     flutter pub run intl_utils:generate
     ```

6. **Test Your Translations**:

   - Run the app and change the device or emulator language to your target language.
   - Verify that all translations appear correctly in the app.

7. **Submit a Pull Request**:

   - Commit your changes and push them to your forked repository.
   - Submit a pull request to the main repository for review.

### Additional Notes

- **Consistency**: Ensure that you maintain the same placeholders and variables used in the English template. For example, if there are variables like `{userName}`, they should be included exactly as is in your translations.
- **Singular and Plural nouns**: Be careful in handling cases where singular and plural have different wordings. For eg: **You are deleting 1 note** and **You are deleting 3 notes**, while in English the difference is just adding `s` for plural, it may be complex for other languages. 
- **Collaboration**: If you're unsure about a translation, consider discussing it with native speakers or opening an issue for community input.

**Comment by rsg-devghost** [comment]
Hey @SankethBK , would like to contribute on this issue.

**Comment by SankethBK** [comment]
@rsg-devghost yes, please go ahead. Just mention what you'll be contributing to avoid conflict

**Comment by rsg-devghost** [comment]
Sure, just a simple doubt , are some of the language translations incomplete?. If so i would like to complete, "Kannada - kn" for instance. Also I would like to add  2 languages "Russian - ru" and "Bengali Bangla - bn".

**Comment by SankethBK** [comment]
Yes, most of them are incomplete.

**Comment by VedicGoyal** [comment]
Hey, I would like to contribute on this.
Also can I get your contact or linkedin, I want to ask some things and want to get insights from you.

**Comment by SankethBK** [comment]
@VedicGoyal you can join our discord group, link is in readme

**Comment by SankethBK** [comment]
Sure you can go ahead @jyk0619
